src/workerd/jsg/util.c++

```diff
@@ -388,6 +421,7 @@ void addExceptionDetail(Lock& js, kj::Exception& exception, v8::Local<v8::Value>
           // Make sure we don't break compatibility if V8 introduces a new version. This value can
           // be bumped to match the new version once all of production is updated to understand it.
           .version = 15,
+          .treatErrorsAsHostObjects = true,
```

Comment:
Reviewer note: One thing that is not sufficiently clear right now is whether this will change the behavior sufficiently enough that a compat flag is needed. In some cases, as illustrated in the modified tests, the `name` and `message` of the resulting error may be different.

Reply:
In some of the cases involving custom error types (e.g. `PythonError`, the name was being added to the message as a prefix, e.g. something like `Error: PythonError {rest of message}`, with this change the name of the error is preserved so it would come through as `PythonError: {rest of message}`.

The other key difference is that own properties on the original errors would now make it through if they were in the js serialized detail. So things like an own property like `retriable: true` would be preserved through the tunnel whereas currently the own properties are dropped.

Reply:
Hmm, I think I'd lean towards not needing a compat flag, because the worst that happens is an already-broken case (an error was thrown) gets a little bit more broken (the reporting gets confused)...